src/routing_utils/nats_client/main.go

```diff
@@ -0,0 +1,193 @@
+package main
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net/url"
+	"os"
+	"time"
+
+	"code.cloudfoundry.org/tlsconfig"
+	"github.com/nats-io/nats.go"
+	"gopkg.in/yaml.v2"
+)
+
+const USAGE = `Usage:
+/var/vcap/jobs/gorouter/bin/nats_client [COMMAND] [SUBJECT] [MESSAGE]
+
+COMMANDS:
+  subscribe    (Default) Streams NATS messages from server with provided SUBJECT. Default SUBJECT is 'router.*'
+               Example: /var/vcap/jobs/gorouter/bin/nats_client subscribe 'router.*'
+
+  publish      Publish the provided message JSON to SUBJECT subscription. SUBJECT and MESSAGE are required
+               Example: /var/vcap/jobs/gorouter/bin/nats_client publish router.register '{"host":"172.217.6.68","port":80,"uris":["bar.example.com"]}'
+`
+
+// Simple NATS client for debugging
+// Uses gorouter.yml for config
+func main() {
+	if os.Args[len(os.Args)-1] == "--help" || os.Args[len(os.Args)-1] == "-h" || os.Args[len(os.Args)-1] == "help" {
+		fmt.Println(USAGE)
+		os.Exit(1)
+	}
+
+	configPath := os.Args[1]
+	command := "subscribe"
+	if len(os.Args) >= 3 {
+		command = os.Args[2]
+	}
+	if command != "subscribe" && command != "publish" {
+		fmt.Println(USAGE)
+		os.Exit(1)
+	}
+
+	subject := "router.*"
+	if len(os.Args) >= 4 {
+		subject = os.Args[3]
+	}
+
+	var message string
+	if command == "publish" {
+		if len(os.Args) >= 5 {
+			message = os.Args[4]
+		} else {
+			fmt.Println(USAGE)
+			os.Exit(1)
+		}
+	}
+
+	config, err := loadConfig(configPath)
+	if err != nil {
+		panic(err)
+	}
+
+	natsOptions, err := natsOptions(config)
+	if err != nil {
+		panic(err)
+	}
+
+	natsConn, err := natsOptions.Connect()
+	if err != nil {
+		panic(err)
+	}
+
+	if command == "publish" {
+		fmt.Fprintf(os.Stderr, "Publishing message to %s\n", subject)
+		err := natsConn.Publish(subject, []byte(message))
+		if err != nil {
+			panic(err)
+		}
+		fmt.Fprintln(os.Stderr, "Done")
+	}
+
+	if command == "subscribe" {
+		fmt.Fprintf(os.Stderr, "Subscribing to %s\n", subject)
+		subscription, err := natsConn.SubscribeSync(subject)
+		if err != nil {
+			panic(err)
+		}
+
+		for {
+			message, err := subscription.NextMsg(time.Second * 60)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "error fetching message: %s", err.Error())
+			} else {
+				fmt.Println(string(message.Data))
```

Comment:
Just stumbled across this while testing:
when you don't subscribe to a subject you will subscribe to `router.*` but here we don't print the subject this message was about, so we don't actually know whether the message is about a route registering or unregistering. 
Should be useful to also print the subject here.

Reply:
Updated :)